ibc/relayer.go

```diff
@@ -82,6 +82,13 @@ type Relayer interface {
 	// Relayer implementations provided by the ibctest module will report true,
 	// but custom implementations may report false.
 	UseDockerNetwork() bool
+
+	// Exec runs an arbitrary command using Relayer's docker environment.
+	// Whether the invoked command is run in a one-off container or execing into an already running container
+	// is up to the relayer implementation.
+	//
+	// "env" are environment variables in the format "MY_ENV_VAR=value"
+	Exec(ctx context.Context, rep RelayerExecReporter, cmd []string, env []string) (stdout, stderr []byte, err error)
```

Comment:
This API is awkward in that it hides the error code inside the error value, and then you would have to have knowledge of the `dockerutil` package to get at the `ContainerExecResult` to get it. That makes me think we may want `ExecResult` to live in the `ibc` package.

`github.com/strangelove-ventures/ibctest/relayer` already imports both `ibc` and `dockerutil`; to avoid introducing a new `ibc`<->`dockertest` dependency, we may need both of those package to have their own `ExecResult` and then the `relayer` package would translate the `dockertest`'s type into the `ibc`'s type.

Reply:
It didn't look like we needed the error code outside of `Exec` since it was only ever being used to pass into the `RelayerExecReporter` & the call to `HandleNodeJobError`. We still pass the error code into the test reporter and we get errors back that are formatted with the error code after the call to `image.Run` which makes `HandleNodeJobError` unused code.
https://github.com/strangelove-ventures/ibctest/blob/cb2a4ec120c97af72f4af6bf6b0c1ad49839def5/relayer/docker.go#L205-L208

If it seems like there could be additional use cases for accessing the error code that are outside of how we previously used it then I can definitely make the changes to return the entire struct by using some adapter like you suggested.

Reply:
I expect that we will eventually have tests that want to assert against relayer exit codes. It is possible those tests won't live in ibctest, but will rather be written by consumers of ibctest. I think we should expose the exit code, via something like `ExecResult`, to support that use case.

Reply:
I'll get those changes made as well!

Reply:
I came back to this PR after two weeks and wasn't sure if the suggestion to have a `ContainerExecResult` type in both `dockerutil` and `relayer` was to avoid a circular dependency, or forward thinking to avoid some issue later down the line. 
 
 I adjusted the API to return the struct vs. it's fields, but if it's still desirable to have a unique type in both packages then I can definitely get that taken care of.